plugins/modules/postgresql_pg_hba.py

```diff
@@ -282,15 +306,27 @@ def read(self):
         try:
             with open(self.pg_hba_file, 'r') as file:
                 for line in file:
+                    # split into line and comment
                     line = line.strip()
-                    # uncomment
+                    comment = None
                     if '#' in line:
                         line, comment = line.split('#', 1)
-                        self.comment.append('#' + comment)
-                    try:
-                        self.add_rule(PgHbaRule(line=line))
-                    except PgHbaRuleError:
-                        pass
+                        if comment == '':
+                            comment = None
+
+                    # if there is just a comment, save it
+                    if line == '':
+                        if comment is not None:
+                            self.comment.append('#' + comment)
```

Comment:
```suggestion
                            self.comment.append('# ' + comment)
```
What do you think?

Reply:
I think this space will break idempotency, because the `comment` variable contains everything of the parsed line behind the `#`.
We could do something like `comment = ' ' + comment` to the `comment` _module parameter_ to insert a space before that comment. But that is a question of style which we are currently leaving up to the user.

Reply:
i think in most cases i say `# comment`, for example in default config files of some services, etc. If possible, i'd be happy to see the space between `#` and the rest.